Source/PLCrashReporter.m

```diff
@@ -608,24 +613,22 @@ - (BOOL) enableCrashReporterAndReturnError: (NSError **) outError {
                 return NO;
             
             /* Enable the server. */
-            _machServer = [self enableMachExceptionServerWithPreviousPortSet: &_previousMachPorts
+            _machServer = [self enableMachExceptionServerWithPreviousPortSet:  &_previousMachPorts
                                                                     callback: &mach_exception_callback
                                                                      context: &signal_handler_context
                                                                        error: outError];
             if (_machServer == nil)
                 return NO;
             
-            /* Acquire references to the autoreleased values */
-            [_machServer retain];
-            [_previousMachPorts retain];
-            
             /*
              * MEMORY WARNING: To ensure that our instance survives for the lifetime of the callback registration,
-             * we retain it here. This is necessary to ensure that the Mach exception server instance and previous port set
+             * we keep a reference on self. This is necessary to ensure that the Mach exception server instance and previous port set
              * survive for the lifetime of the callback. Since there's currently no support for *deregistering* a crash reporter,
              * this simply results in the reporter living forever.
+             * We do not check pl_crash_reporter nullability here, because we already ensured
+             * that this is a single instance.
              */
-            [self retain];
+            pl_crash_reporter = self;
```

Comment:
`CFBridgingRetain()` (with a corresponding `CFBridgingRelease()`) would be more appropriate here, and would avoid introducing additional global singleton state.

Reply:
Good to see you here. Thank you for following the changes

Reply:
@landonf I don't think `CFBridgingRelease()` is necessary here. Prior to our changes there was a single `[self retain]` here - without any explicit releasing (and the explanation was in the comment). 
So I think it makes sense to just replace it with `CFBridgingRetain()`.